runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaMergedFactory.java

```diff
@@ -1316,6 +1328,23 @@ private void onMergedInitialFlush(
                 doFetchPartitionsIfNecessary(traceId);
                 doProducePartitionsIfNecessary(traceId);
             }
+            else
+            {
+                if (hasProduceCapability(capabilities))
+                {
+                    final KafkaOffsetFW partition = kafkaMergedFlushEx.partition();
+                    final KafkaKeyFW key = kafkaMergedFlushEx.key();
+                    if (partition != null)
+                    {
+                        final int partitionId = partition.partitionId();
+                        final int nextPartitionId = partitionId == DYNAMIC_PARTITION ? nextPartitionFlush(key) : partitionId;
```

Comment:
Since `FLUSH` needs to be propagated to a specific `partition`, wouldn't it make sense to require either `key` or  explicit `partitionId` rather than supporting dynamic partition approach used for `DATA`?

Reply:
As discussed, I'll leave it as is.